src/Features/CSharp/Portable/UseLocalFunction/CSharpUseLocalFunctionDiagnosticAnalyzer.cs

```diff
@@ -106,10 +106,10 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext, INamedTyp
                 return;
             }
 
-            // If there are compiler diagnostics on the declaration we can't reliably
+            // If there are compiler error on the declaration we can't reliably
             // tell that the refactoring will be accurate, so don't provide any
             // code diagnostics
-            if (localDeclaration.GetDiagnostics().Any())
+            if (localDeclaration.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
```

Comment:
You could use `DefaultSeverity` instead to filter out diagnostics whose severity is changed by user. But I'm fine either way here.